### PR TITLE
Add all-namespaces flag to svcat

### LINKED ISSUE #1766: Support listing across all namespaces in svcat
Just like kubectl, we should make it easier to list instances and bindings across namespaces. Right now you can do this with `svcat get instances -n=""`.  The kubectl way would be `svcat get instances --all-namespaces`.

### PR DESCRIPTION
**Pull request by cgmcintyr**
* Closes #1766

* Add all-namespaces flag to svcat get bindings/instances commands

When set retrieves all bindings/instances across namespaces, overriding
the value of the ns flag.

* Add tests for all-namespaces flag

Add testdata and tests for both `svcat get bindings --all-namespaces`
and `svcat get instances --all-namespaces`

### CODE REVIEW

**Comment by carolynvs** [comment]
I can't wait to use this, thanks for tackling it! 💖

File: cmd/svcat/instance/get_cmd.go [hunk 1]
```diff
@@ -37,6 +38,7 @@ func NewGetCmd(cxt *command.Context) *cobra.Command {
 		Short:   "List instances, optionally filtered by name",
 		Example: `
   svcat get instances
+  svcat get instances --all-namespaces
```
Comment (n3wscott) [inline cmd/svcat/instance/get_cmd.go]: This feels like a pretty verbose flag, what if we make the default for `-n` to be "" and in a sense, the lack of specifying a namespace would be all namespaces. If you want the default namespace you have to say `-n default`.

Comment (n3wscott) [reply cmd/svcat/instance/get_cmd.go]: What say you @carolynvs ?

Comment (carolynvs) [reply cmd/svcat/instance/get_cmd.go]: That would be a behavior difference on a shared component between svcat and kubectl which we avoid to maintain consistent behavior between using svcat alone and as a kubectl plugin.

When a namespace is not set in the current context, kubectl uses "default", and to query all namespaces, kubectl has the `--all-namespaces` flag. This PR brings svcat in-line with kubectl, and enables us to respect the `--all-namespaces` flag when it is passed to us by kubectl in plugin mode.

Once this goes in, you can continue to use `-n=""` to get all namespaces if you prefer a shorter syntax. 😁

Comment (jberkhahn) [reply cmd/svcat/instance/get_cmd.go]: Yeah, I would stick with the behavior kubectl uses as well - and it uses --all-namespaces for several commands.

Comment (n3wscott) [reply cmd/svcat/instance/get_cmd.go]: ok thanks for the clarification!!

**Pull request merged**